### PR TITLE
Add Event Hub Stream binders Health IT

### PR DESCRIPTION
**Pull request by wujack778**
Add integration test for EH binder health indicator
[issue link](https://dev.azure.com/azure-spring-integration/spring-integration-private/_sprints/taskboard/spring-integration-private%20Team/spring-integration-private/186?workitem=623)
[Add integration test for EH binder health indicator](https://dev.azure.com/azure-spring-integration/spring-integration-private/_sprints/taskboard/spring-integration-private%20Team/spring-integration-private/186?workitem=1088)

### CODE REVIEW

File: sdk/spring/azure-spring-cloud-test-eventhubs/pom.xml [hunk 1]
```diff
@@ -19,6 +19,15 @@
   <description>EventHubs Test project for Spring Cloud</description>
 
   <dependencies>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+    </dependency>
+    <dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-eventhubs/pom.xml:26]: why is the storage dependency needed?

File: sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import static com.azure.spring.integration.core.AzureHeaders.CHECKPOINTER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+public class HealthBindersIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HealthBindersIT.class);
+
+    private RestTemplate REST_TEMPLATE = new RestTemplate();
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:29]: Do NOT use this style `REST_TEMPLATE` if the variable is not static.

Comment (saragluna) [reply sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:29]: Do use restTemplate.

File: sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            app.property("spring.cloud.stream.bindings.consume-in-0.destination", "event-hub-name");
+            app.property("spring.cloud.stream.bindings.consume-in-0.group", "$Default");
+            app.property("spring.cloud.stream.bindings.supply-out-0.destination", "event-hub-name");
+            app.property("spring.cloud.stream.default.producer.errorChannelEnabled", "true");
+            app.property("spring.cloud.stream.eventhub.bindings.consume-in-0.consumer.checkpoint-mode", "MANUAL");
+            app.property("spring.cloud.stream.function.definition", "consume;supply");
+
+            app.property("management.endpoint.health.show-details", "always");
+            app.property("management.endpoints.web.exposure.include", "*");
+            app.property("management.health.binders.enabled", "true");
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:46]: I remember the default value of `management.health.binders.enabled` is `true`.

File: sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    message.getHeaders().get(EventHubHeaders.OFFSET),
+                    message.getHeaders().get(EventHubHeaders.ENQUEUED_TIME)
+                );
+
+                checkpointer.success()
+                            .doOnSuccess(success -> LOGGER.info("Message '{}' successfully checkpointed",
+                                message.getPayload()))
+                            .doOnError(error -> LOGGER.error("Exception found", error))
+                            .subscribe();
+            };
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:83]: This is too complicated for a test case, maybe below is enough?
```java
message -> {
    LOGGER.info("Message received: {}", message);
}
```

File: sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        @ServiceActivator(inputChannel = "{destination}.{group}.errors")
+        public void consumerError(Message<?> message) {
+            LOGGER.error("Handling customer ERROR: " + message);
+        }
+
+        // Replace destination with spring.cloud.stream.bindings.supply-out-0.destination
+        @ServiceActivator(inputChannel = "{destination}.errors")
+        public void producerError(Message<?> message) {
+            LOGGER.error("Handling Producer ERROR: " + message);
+        }
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:97]: We don't these for this test case, right?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java (hunk 5 above)
Comment (wujack778) [reply sdk/spring/azure-spring-cloud-test-eventhubs/src/test/java/com/azure/spring/eventhubs/health/HealthBindersIT.java:97]: get it

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**